copy/copy.go

```diff
@@ -1166,6 +1166,18 @@ func computeDiffID(stream io.Reader, decompressor compression.DecompressorFunc)
 	return digest.Canonical.FromReader(stream)
 }
 
+type streamReader struct {
+	reader io.Reader
+}
+
+func (r streamReader) Read(b []byte) (n int, err error) {
+	n, err = r.reader.Read(b)
+	if err != nil {
+		return n, errors.Wrapf(err, "error happened during read")
```

Comment:
Oops, looking at tests, this must not annotate `io.EOF`, because exactly that value is a part of the `io.Reader` protocol.